Mccalluc/clinical imaging

mccalluc opened the pull request:
Fix #1081

Filing as draft: There are lots of TODOs that need clarification -- I won't repeat them there. (Better to start review from the bottom, in `src/`, and skip `docs/`.)

@ngehlenborg : Assigning to you for review by default. I have gotten the sense that you want to be involved in these decisions, but I also get the sense that you don't have the time to be involved. Articulate the process you want in this case, and I'll try to follow it.

One high-level question is the names of these assays, in particular "Micro CT", "OCT", "Body CT" ... space or no space? what does "O" stand for? (When it's settled, should ping @jswelling to make sure they make it to the list he maintains.)

I don't think there's a TODO for this, but can you confirm that removing `analyte_class` and `is_targeted` makes sense for these? We have said these ("level 1") fields will be on all assays, but they just aren't applicable here.

mccalluc commented:
All the TODO questions:
- [ ] Are the assay names in the PR correct? I have made changes from the spreadsheet for consistentency in spacing and capitalization.
- [ ] For each assay, what is the URL for the complete documentation? Should be under https://software.docs.hubmapconsortium.org/
- [ ] For each assay, what is a sentance or two that should appear on the ingest-validation-tools page for the assay?
- [ ] What is the description for `single_file_export_format`?
- [ ] Body CT, unlike the other assays, does not have any fields particular to it. Confirm that this correct?
- [ ] Confirm that "kVa" is correct?
- [ ] Confirm that "V" can be used for "Volt", for consistency? (We have not spelled out unit abbreviations in other assays.)
- [ ] Confirm that "Amp" should be spelled out in "mAmp", "microAmp",  "Amp"? (Though my preference would be consistency.)
- [ ] Confirm that it's ok to replace "imaging_threshold_unit" with "imaging_threshold_unit_value"? (Other "_unit" fields are enums.)
- [ ] Confirm that "echo_time" and "repetition_time" and "flip_angle" do not need an explicit unit?
- [ ] Confirm that interger is correct for "number_of_averages"?
- [ ] Confirm that the description "The number of sections used for analyzing microCT images" is ok to include in OCT?
- [ ] Confirm that "high" is the only allowed value for "quality_view"?
- [ ] Should this wait for the corresponding directory schema, or should it be released on its own?

ngehlenborg commented:
Final list of questions sent to clinical imaging team on 5/6/2022:

- [ ] 1. Are these assay names correct? We have made changes from the spreadsheet for consistency in spacing and capitalization:
Body CT
Micro CT
MRI
OCT
Ultrasound

- [ ] 2. For each assay, what is the URL for the complete documentation? Should be under [https://software.docs.hubmapconsortium.org/assays.html](https://software.docs.hubmapconsortium.org/)

- [ ] 3. For each assay, what is a sentence or two that should appear on the ingest-validation-tools page for the assay? Please explain any acronyms! For examples see here: https://hubmapconsortium.github.io/ingest-validation-tools/, e.g. https://hubmapconsortium.github.io/ingest-validation-tools/stained/ ("This schema is for microscopy of tissue treated with periodic acid–Schiff stain (stained). […]”)

- [ ] 4. What is the description for `single_file_export_format`?

- [ ] 5. Body CT, unlike the other assays, does not have any fields particular to it. Confirm that this correct?

- [ ] 6. Confirm that "kVa" is correct?

- [ ] 7. Confirm that "V" can be used for "Volt", for consistency? (We have not spelled out unit abbreviations in other assays.)

- [ ] 8. Confirm that "Amp" should be spelled out in "mAmp", "microAmp",  "Amp"? (Though my preference would be consistency.)

- [ ] 9. Confirm that it's ok to replace "imaging_threshold_unit" with "imaging_threshold_unit_value"? (Other "_unit" fields are enums.)

- [ ] 10. Confirm that "echo_time" and "repetition_time" and "flip_angle" do not need an explicit unit?

- [ ] 11. Confirm that integer is correct type for "number_of_averages”?

- [ ] 12. Confirm that the description "The number of sections used for analyzing microCT images" is ok to include in OCT or provide alternative description.

- [ ] 13. Confirm that "high" is the only allowed value for "quality_view”?

- [ ] 14. Should the metadata schemas be released as soon as possible or should we wait for the corresponding directory schemas?

- [ ] 15. Confirm that these all are "Imaging" assays, rather than belonging to a new category?All the TODO questions:

mccalluc commented:
Answers!

> 1. Are these assay names correct? Yes, but they’re imaging modalities, not assays. The overall assay is “clinical imaging”.  We have made changes from the spreadsheet for consistency in spacing and capitalization: Thanks!
> 
>  
> 
> Body CT
> 
> Micro CT
> 
> MRI
> 
> OCT
> 
> Ultrasound
> 
>  
> 
> 2. For each assay, what is the URL for the complete documentation? Should be under [https://software.docs.hubmapconsortium.org/assays.html](https://urldefense.com/v3/__https:/software.docs.hubmapconsortium.org/__;!!LLK065n_VXAQ!hBDDOXbO9-3-hk18HDA5FiOgQl3-GXzwpIGnAs82Iijwn1KG1Mt9Ht0-Z_eYtdaEwQw2DZ041t4xL-hezUXO1A$)
> 
> We are documenting all 5 imaging modalities under this single “Clinical Imaging” [assay document](https://urldefense.proofpoint.com/v2/url?u=https-3A__docs.google.com_document_d_1P0dKc7CzfblBANhPDz0sql0XtztEJja6_edit&d=DwMGaQ&c=WO-RGvefibhHBZq3fL85hQ&r=-zahbihG7P2H0fiPo3RTqce9mFyFku-NJy5aRYNyUSA&m=hEt4h2R0xjo2MjV05ZsHDnBQL_2zbXYR2NPJjm04u5KF0b3le3cCE-0YHyZjWUcs&s=JCzWqrJ3XSkFA5YEnckgL-1F1DqNbPQGQyrDcVoyWmA&e=). It’s not yet posted on the site you provided since we’re still trying to finalize it (by adding the link to the metadata specs you’re currently reviewing and finalizing the directory structure with Ivan).
> 
>  
> 
> 3. For each assay, what is a sentence or two that should appear on the ingest-validation-tools page for the assay? Please explain any acronyms! For examples see here: [https://hubmapconsortium.github.io/ingest-validation-tools/](https://urldefense.com/v3/__https:/hubmapconsortium.github.io/ingest-validation-tools/__;!!LLK065n_VXAQ!hBDDOXbO9-3-hk18HDA5FiOgQl3-GXzwpIGnAs82Iijwn1KG1Mt9Ht0-Z_eYtdaEwQw2DZ041t4xL-jppaSk4w$), e.g. [https://hubmapconsortium.github.io/ingest-validation-tools/stained/](https://urldefense.com/v3/__https:/hubmapconsortium.github.io/ingest-validation-tools/stained/__;!!LLK065n_VXAQ!hBDDOXbO9-3-hk18HDA5FiOgQl3-GXzwpIGnAs82Iijwn1KG1Mt9Ht0-Z_eYtdaEwQw2DZ041t4xL-heoDZQQA$) ("This schema is for microscopy of tissue treated with periodic acid–Schiff stain (stained). […]”)
> 
> Please use the following sentence. Again, all imaging modalities listed above will be contained a single metadata schema document for the clinical imaging assay: 
> 
> “This schema is for clinical imaging modalities using any of the following different technologies: body computer tomography (Body CT), micro computer tomography (Micro CT), optical coherence tomography (OCT), ultrasound, or magnetic resonance imaging (MRI).”
> 
>  
> 
> 4. What is the description for `single_file_export_format`?
> 
> “The format in which each single imaging file will be exported (Example: DICOM, tiff, avi, etc)” I have added this to the metadata google sheet.
> 
>  
> 
> 5. Body CT, unlike the other assays, does not have any fields particular to it. Confirm that this correct?
> 
> Yes, this is correct.
> 
>  
> 
> 6. Confirm that "kVa" is correct?
> 
> If this is for row 33 of the metadata table? It should be kV instead of kVp. I corrected to kV.
> 
>  
> 
> 7. Confirm that "V" can be used for "Volt", for consistency? (We have not spelled out unit abbreviations in other assays.)
> 
> If you prefer to stay consistent with other assay, we don’t have to use abbreviations. But yes, V can be used for Volt.
> 
>  
> 
> 8. Confirm that "Amp" should be spelled out in "mAmp", "microAmp",  "Amp"? (Though my preference would be consistency.)
> 
> If we’re using V for Volt we can abbreviate this to use A for Amp, so it would be mA, microA, A.
> 
>  
> 
> 9. Confirm that it's ok to replace "imaging_threshold_unit" with "imaging_threshold_unit_value"? (Other "_unit" fields are enums.)
> 
> Yes, it’s ok to replace this.
> 
>  
> 
> 10. Confirm that "echo_time" and "repetition_time" and "flip_angle" do not need an explicit unit?
> 
> Yes, confirmed. For all three the unit is defined in the description.
> 
>  
> 
> 11. Confirm that integer is correct type for "number_of_averages”?
> 
> We have this as float, not integer, and prefer to keep it as float.
> 
>  
> 
> 12. Confirm that the description "The number of sections used for analyzing microCT images" is ok to include in OCT or provide alternative description.
> 
> Thanks for catching this, it should be corrected to: “The number of sections used for analyzing microCT or OCT images”
> 
>  
> 
> 13. Confirm that "high" is the only allowed value for "quality_view”?
> 
> I see these enums for quality_view (row 45): high confidence/optimal, low confidence/sub-optimal, no confidence
> 
>  
> 
> 14. Should the metadata schemas be released as soon as possible or should we wait for the corresponding directory schemas?
> 
> Yes please! We thought they were already released in March. And we are told we need to github link to add to our directory schema so it seems metadata should be released first.
> 
>  
> 
> 15. Confirm that these all are "Imaging" assays, rather than belonging to a new category?
> 
> Yes, these are all various imaging modalities belonging to the “clinical imaging assay” category.

mccalluc commented:
Comments during implementation:

>Yes, but they’re imaging modalities, not assays. The overall assay is “clinical imaging”.

This seems important to them, but I’m not sure what action to take… Should there be a separate section for these 5 in the TOC on gh-pages? Right now there among other “Imaging assays”





>Please use the following sentence. Again, all imaging modalities listed above will be contained a single metadata schema document for the clinical imaging assay:
>“This schema is for clinical imaging modalities using any of the following different technologies: body computer tomography (Body CT), micro computer tomography (Micro CT), optical coherence tomography (OCT), ultrasound, or magnetic resonance imaging (MRI).”

There’s some confusion: It’s 5 separate schemas, not a single schema that covers all five. I’ll make the descriptions follow this pattern: “This schema is for clinical imaging using body computer tomography (Body CT)” … except it should be “computed”, not “computer”.

```
single_file_export_format
  description: ’The format in which each single imaging file will be exported. (Example: DICOM, tiff, avi
```
Seems like there could be an enum, but not going to worry about that at this point.

microA … for length, we’ve used um… but I think micro is fine here, if that’s what they suggest. 

>> Confirm that these all are “Imaging” assays, rather than belonging to a new category?
>
> Yes, these are all various imaging modalities belonging to the “clinical imaging assay” category.

Ok, I read that as saying that they want a new category…. Not sure exactly what the implementation will look like…

Ok: Introducing a new  assay_category type. This is significant enough that I will ask for review on the PR after all.

ngehlenborg reviewed (approved):
Thanks for a careful implementation!
